src/rules/convertComments.ts

```diff
@@ -0,0 +1,74 @@
+import { FileSystem } from "../adapters/fileSystem";
+import { isError } from "../utils";
+import { getCommentAtPosition } from "tsutils";
+import * as ts from "typescript";
+
+// Check https://github.com/Microsoft/TypeScript/issues/21049
+
+export type ConvertCommentsResultsDependencies = {
+    fileSystem: Pick<FileSystem, "readDir" | "readFile" | "writeFile">;
+};
+
+export const convertComments = async (dependencies: ConvertCommentsResultsDependencies) => {
+    // TODO: Remove console logs
+    console.log("Started");
+    const filenames = await dependencies.fileSystem.readDir("./", { withFileTypes: true });
+    if (!isError(filenames)) {
+        const filteredFilenames: string[] = filenames
+            .filter(fileEnt => fileEnt.isFile())
+            .map(fileEnt => fileEnt.name);
+        // TODO: Remove console logs
+        console.log("Filenames filtered");
+        console.log(filteredFilenames);
+        for (const filename of filteredFilenames) {
+            const fileContent: string | Error = await dependencies.fileSystem.readFile(filename);
+            if (!isError(fileContent)) {
+                const writeFileRes = await replaceComments(dependencies, filename, fileContent);
+                if (isError(writeFileRes)) {
+                    return Error("Failed to convert file comments");
+                }
+            }
+        }
+        return undefined;
+    } else {
+        return Error("Failed to convert file comments");
+    }
+};
+
+const replaceComments = async (
+    dependencies: ConvertCommentsResultsDependencies,
+    fileName: string,
+    fileContent: string,
+) => {
+    const tslintRegex: RegExp = new RegExp(
+        /\/[\/\*]\s*(tslint):(enable|disable)((?:-next)?-line)?(?::\s)?(:?(?:(?:[\w-]+\/)*[\w-]+\s*\s*)*(?:(?:[\w-]+\/)*[\w-]+\s*)*[\w-]+)?\s*?(?:$|\*\/)/gm,
```

Comment:
My eyes are bleeding. 🙀 

The vast majority of comments that start with `(tslint):(enable|disable)` will be valid. How about just flagging for that, then using traditional string comparisons to check the rest?

Reply:
Oh, even better! `tslint` exports a `ENABLE_DISABLE_REGEX` just for this.

Reply:
> Oh, even better! `tslint` exports a `ENABLE_DISABLE_REGEX` just for this.

OMG! Just saw the comment and went into google to find out that it is a short (and much cleaner) regex pattern to work on... Again, not the most skilled regex guy around but did my best 😆 

However, need to test it first so I can retrieve the rule group to split it into an array of strings to loop each one on the rule dictionary and provide the proper replacement, that should be for the rule replacement part.

Reply:
To be fair, I do think we got some stuff for the missing conversion logic at the [tslint source code](https://github.com/palantir/tslint/blob/master/src/enableDisableRules.ts).